db/write_stall_stats.cc

```diff
@@ -0,0 +1,154 @@
+          break;
+      }
+      break;
+    }
+    case WriteStallCause::kL0FileCountLimit: {
+      switch (condition) {
+        case WriteStallCondition::kDelayed:
+          return InternalStats::L0_FILE_COUNT_LIMIT_DELAYS;
+        case WriteStallCondition::kStopped:
+          return InternalStats::L0_FILE_COUNT_LIMIT_STOPS;
+        case WriteStallCondition::kNormal:
+          break;
+      }
+      break;
+    }
+    case WriteStallCause::kPendingCompactionBytes: {
+      switch (condition) {
+        case WriteStallCondition::kDelayed:
+          return InternalStats::PENDING_COMPACTION_BYTES_LIMIT_DELAYS;
+        case WriteStallCondition::kStopped:
+          return InternalStats::PENDING_COMPACTION_BYTES_LIMIT_STOPS;
+        case WriteStallCondition::kNormal:
+          break;
+      }
+      break;
+    }
+    default:
+      break;
+  }
+  return InternalStats::INTERNAL_CF_STATS_ENUM_MAX;
+}
+
+InternalStats::InternalDBStatsType InternalDBStat(
+    WriteStallCause cause, WriteStallCondition condition) {
+  switch (cause) {
+    case WriteStallCause::kWriteBufferManagerLimit: {
+      switch (condition) {
+        case WriteStallCondition::kStopped:
+          return InternalStats::kIntStatsWriteBufferManagerLimitStopCounts;
+        default:
+          break;
+      }
+      break;
+    }
+    default:
+      break;
+  }
+  return InternalStats::kIntStatsNumMax;
+}
+
+bool isCFScopeWriteStallCause(WriteStallCause cause) {
+  uint32_t int_cause = static_cast<uint32_t>(cause);
+  uint32_t lower_bound =
+      static_cast<uint32_t>(WriteStallCause::kCFScopeWriteStallCauseEnumMax) -
+      kNumCFScopeWriteStallCauses;
+  uint32_t upper_bound =
+      static_cast<uint32_t>(WriteStallCause::kCFScopeWriteStallCauseEnumMax) -
+      1;
+  return lower_bound <= int_cause && int_cause <= upper_bound;
+}
+
+bool isDBScopeWriteStallCause(WriteStallCause cause) {
+  uint32_t int_cause = static_cast<uint32_t>(cause);
+  uint32_t lower_bound =
+      static_cast<uint32_t>(WriteStallCause::kDBScopeWriteStallCauseEnumMax) -
+      kNumDBScopeWriteStallCauses;
+  uint32_t upper_bound =
+      static_cast<uint32_t>(WriteStallCause::kDBScopeWriteStallCauseEnumMax) -
+      1;
+  return lower_bound <= int_cause && int_cause <= upper_bound;
+}
+
+const std::string& WriteStallStatsMapKeys::TotalStop() {
+  static const std::string kTotalStop = "total-stop";
+  return kTotalStop;
+}
+
+const std::string& WriteStallStatsMapKeys::TotalDelay() {
+  static const std::string kTotalDelay = "total-delay";
+  return kTotalDelay;
+}
+
+const std::string&
+WriteStallStatsMapKeys::CFL0FileCountLimitDelaysWithOngoingCompaction() {
+  static const std::string ret =
+      "cf-l0-file-count-limit-delays-with-ongoing-compaction";
+  return ret;
+}
+
+const std::string&
+WriteStallStatsMapKeys::CFL0FileCountLimitStopsWithOngoingCompaction() {
+  static const std::string ret =
+      "cf-l0-file-count-limit-stops-with-ongoing-compaction";
```

Comment:
"stops"

The point of this and the above two comments isn't to increase my comment count with low effort, but to point out the terminology changes slightly throughout the file. Can we use "stops" in all the places (and "delays")?

Reply:
sounds good to me!

Reply:
Fixed